Agents/DevInstAgent/DevInstAgent/src/test/java/uk/ac/cam/cares/jps/agent/devinst/DevInstQureyBuilderTest.java

```diff
@@ -0,0 +1,352 @@
+import org.json.JSONObject;
+import org.junit.*;
+import org.junit.rules.TemporaryFolder;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mockito;
+import org.semanticweb.owlapi.util.IRIComparator;
+
+import com.github.stefanbirkner.systemlambda.SystemLambda;
+
+import uk.ac.cam.cares.jps.base.exception.JPSRuntimeException;
+import uk.ac.cam.cares.jps.base.query.RemoteStoreClient;
+import wiremock.com.jayway.jsonpath.internal.function.text.Length;
+import wiremock.org.eclipse.jetty.util.ajax.JSON;
+
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.junit.jupiter.Container;
+import org.testcontainers.utility.DockerImageName;
+
+import java.io.*;
+import java.lang.reflect.Array;
+import java.net.URISyntaxException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.*;
+
+public class DevInstQureyBuilderTest {
+    private static final String ONTODEV = "https://www.theworldavatar.com/kg/ontodevice/";
+    private static final Prefix P_DEV = SparqlBuilder.prefix("ontodevice",iri(ONTODEV));
+    private static final Prefix P_SAREF = SparqlBuilder.prefix("saref", iri("https://saref.etsi.org/core/"));
+    private static final Prefix P_AGENT = SparqlBuilder.prefix("ontoagent",iri("https://www.theworldavatar.com/kg/ontoagent/"));
+    private static final Prefix P_OM = SparqlBuilder.prefix("om", iri("http://www.ontology-of-units-of-measure.org/resource/om-2/"));
+
+    private static final Iri SmartSensor = P_DEV.iri("SmartSensor");
+    private static final Iri MicroController = P_DEV.iri("MicroController");
+    private static final Iri Sensor = P_SAREF.iri("Sensor");
+    private static final Iri ProximitySensor = P_DEV.iri("ProximitySensor");
+
+    private static final Iri consistsOf = P_SAREF.iri("consistsOf");
+    private static final Iri sendsSignalTo = P_DEV.iri("sendsSignalTo");
+    private static final Iri measures = P_DEV.iri("measures");
+    private static final Iri Task = P_SAREF.iri("Task");
+    private static final Iri hasCommand = P_SAREF.iri("hasCommand");
+    private static final Iri Command = P_SAREF.iri("Command");
+    private static final Iri accomplishes = P_SAREF.iri("accomplishes");
+    // Temporary folder to place a properties file
+    @Rule
+    public TemporaryFolder folder = new TemporaryFolder();
+
+    @Container
+	private GenericContainer<?> blazegraph = new GenericContainer<>(DockerImageName.parse("ghcr.io/cambridge-cares/blazegraph_for_tests:1.0.0"))
+													 .withExposedPorts(9999);
+
+    private DevInstQueryBuilder queryBuilder;
+    RemoteStoreClient storeClient;
+    RemoteStoreClient ontodevClient;
+    RemoteStoreClient sarefClient;
+    
+    JSONObject exampleRequest;
+    JSONObject exampleFind;
+    JSONObject exampleEmptyMap;
+    JSONObject exampleAddQuery;
+
+    //Blazegraph endpoint
+    private String sparql_endpoint;
+    private String ontodev_endpoint = "http://10.25.188.58:3838/blazegraph/namespace/ontodevice/sparql";
```

Comment:
Not a good idea to use a server's IP address for unit testing as these endpoints should be something that is set up and self contained within the tests itself.